flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/IOManagerAsync.java

```diff
@@ -126,22 +131,20 @@ public void close() {
 			// close writing and reading threads with best effort and log problems
 			// first notify all to close, then wait until all are closed
 
-			for (WriterThread wt : writers) {
-				try {
-					wt.shutdown();
-				}
-				catch (Throwable t) {
-					LOG.error("Error while shutting down IO Manager writer thread.", t);
-				}
-			}
-			for (ReaderThread rt : readers) {
-				try {
-					rt.shutdown();
-				}
-				catch (Throwable t) {
-					LOG.error("Error while shutting down IO Manager reader thread.", t);
-				}
-			}
+			Iterable<AutoCloseable> writerThreadCloseables = Arrays.stream(writers)
+				.filter(Objects::nonNull)
+				.filter(Thread::isAlive)
```

Comment:
do we need `Objects::nonNull` and `Thread::isAlive` filters here?

Reply:
`nonNull` is not necessary, `isAlive` might make sense to avoid the thread was already dead unexpected. WDYT?